### DISCUSSION TITLE
Failure in exists query for JavaDB

### DISCUSSION THREAD

**Discussion by DaspawnW** [aquasecurity/trivy#4594] [Bugs]
### Description

During a trial to implement a hack to get the Java DB as a dedicated server #3560 I detected that the signature of the Exists method doesn't match with the SQL Query that is performed. I'm not a golang developer so maybe my bug report is wrong, but I also so far found no description that proofs that I'm wrong.

### Desired Behavior

The Exists function in the [javadb/client.go](https://github.com/aquasecurity/trivy/blob/main/pkg/javadb/client.go#L119) file calls the `SelectIndexByArtifactIDAndGroupID(artifactID, groupID)` out of the [trivy-java-db](https://github.com/aquasecurity/trivy-java-db/blob/main/pkg/db/db.go#L162) in the argument order artifactID, groupID.

### Actual Behavior

The current behavior is that the function gets called by [javadb/client.go](https://github.com/aquasecurity/trivy/blob/main/pkg/javadb/client.go#L119) in the opposite order with groupID, artifactID.

### Reproduction Steps

```bash
1. Compare https://github.com/aquasecurity/trivy/blob/main/pkg/javadb/client.go#L119 with https://github.com/aquasecurity/trivy-java-db/blob/main/pkg/db/db.go#L162
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
nn
```

### Operating System

nn

### Version

```bash
latest main branch
```

### Checklist

- [ ] Run `trivy --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by afdesk** [comment]
@DaspawnW thanks for the report! 
created #4620
